web_project/blog/tasks.py

```
@@ -124,30 +124,29 @@ def generate_ai_text(self, prompt):
     def generate_ai_image(self, prompt):
         api_key = self.creds["AI_IMAGE"]
         r = requests.post(
-        "https://api.deepai.org/api/text2img",
-        data={
-            'text': f'create a header image for this news, make it realistic: {prompt}', # Prompt can be changed
-        },
-        headers={'api-key': api_key}
+            "https://api.deepai.org/api/text2img",
+            data={
+                'text': f'create a header image for this news, make it realistic: {prompt}',
+            },
+            headers={'api-key': api_key}
         )
-        url = f"{r.json()['share_url']}"
         
-        if url:
-            output_dir = os.path.join(settings.MEDIA_ROOT, 'ai_images')
-            os.makedirs(output_dir, exist_ok=True)
+        if r.status_code == 200:
```

check out fail early method for if-else statements. This allows to minimize nested if-else statements.
instead of 
```
if (right):
  do task
if (not right):
  do default / fail
```
do this
```
if (not right): 
  do default / fail
do task
```
check out the resource in here for more info:
https://docs.google.com/document/d/1QD6GLuVkEq1rIgCE5OMCSpP-IK-V6Q8kikACv-tlhAI/edit?tab=t.0